Improve responsiveness of /api/[apiID] page

revogabe opened the pull request:
## What does this PR do?

I adjusted the mobile responsiveness of the /api/[apiID] page. I also updated the filter and calendar buttons to improve the mobile experience by replacing popovers (which were overflowing the screen) with more suitable components.

Additionally, I stacked the charts for better visibility on smaller screens and made a few other minor UI adjustments.

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

Fixes #3020 

*If there is not an issue for this, please create one first. This is used to tracking purposes and also helps use understand why this PR exists*

<!-- If there isn't an issue for this PR, please re-review our Contributing Guide and create an issue -->

## Type of change

<!-- Please mark the relevant points by using [x] -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [X] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration -->

- Check the mobile responsiveness of the `/api/[apiID]` page, especially when there is data/logs to analyze.

> **Note:** I wasn’t able to successfully mock fake data using the log population guide: [https://engineering.unkey.com/contributing/populating-logs](https://engineering.unkey.com/contributing/populating-logs)

## Checklist

<!-- We're starting to get more and more contributions. Please help us making this efficient for all of us and go through this checklist. Please tick off what you did  -->

### Required

- [X] Filled out the "How to test" section in this PR
- [X] Read [Contributing Guide](./CONTRIBUTING.md)
- [X] Self-reviewed my own code
- [X] Commented on my code in hard-to-understand areas
- [X] Ran `pnpm build`
- [X] Ran `pnpm fmt`
- [X] Checked for warnings, there are none
- [X] Removed all `console.logs`
- [X] Merged the latest changes from main onto my branch with `git pull origin main`
- [X] My changes don't cause any responsiveness issues

### Appreciated

https://github.com/user-attachments/assets/bc308408-4347-4de1-82d1-f81e2c575cdf

- [X] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Unkey Docs if changes were necessary

chronark commented:
1. submitting a filter should close the drawer 
https://share.cleanshot.com/FmZJW7Fp

2. changing between time range and calendar has some annoying layout shift 
https://share.cleanshot.com/w41F0KRJ

3. there seem to be some weird scroll bars
![CleanShot 2025-04-09 at 12 24 26@2x](https://github.com/user-attachments/assets/bd09f95c-99a6-4406-b6c6-5e414a9cfb4f)

4. the table is gone?
https://share.cleanshot.com/cx1gkg3T

5. I'm not a fan of the drawer design, let me see if we can cook up something with our designer
![CleanShot 2025-04-09 at 12 28 57@2x](https://github.com/user-attachments/assets/798fa11b-fde9-48a2-81b5-c8dc8dc9b076)

chronark commented:
check this out, we just made the seeding easier and it should work as documented now
https://engineering.unkey.com/contributing/seeding-db-and-clickhouse

revogabe commented:
Sorry for the delay, I was out of town. :sweat_smile: 

I fixed the table that was being hidden and also adjusted the item details when a table row was clicked the details card was breaking. I added an absolute positioned element covering the entire page to make it more visually pleasant for the user.

https://github.com/user-attachments/assets/20822008-e970-47f5-a214-0c98d4d089d3

cc: @chronark

chronark commented:
cool!

should we work on the mobile design to replace the table in a different PR then?

revogabe commented:
what do you want with "replace the table", make a responsive table for mobile? tables are usually scrollable horizontally in most apps when they are on mobile

chronark commented:
> what do you want with "replace the table", make a responsive table for mobile? tables are usually scrollable horizontally in most apps when they are on mobile

This is what I mean, taken from the original issue discussion of this PR
![image](https://github.com/user-attachments/assets/999be666-c0df-4a77-92af-e64e8b1011d1)

revogabe commented:
i see now, yeah it's makes sense, do this in another PR

chronark commented on apps/dashboard/components/logs/checkbox/filters-popover.tsx:
```diff
@@ -17,15 +27,19 @@ type FiltersPopoverProps = {
   getFilterCount = (field) => activeFilters.filter((f) => f.field === field).length,
 }: PropsWithChildren<FiltersPopoverProps>) => {
-  const [open, setOpen] = useState(false);
+  const isMobile = useMediaQuery("(max-width: 768px)");
```
having these magic values is a footgun I think
we'd need to define them in every component, that we make mobile ready, and it'll become a mess.

if we want to use media queries, we should have that hook (and the pixel value) defined in a central place

chronark commented on apps/dashboard/app/(app)/apis/[apiId]/_overview/components/controls/components/logs-filters/outcome-filter.tsx:
```diff
@@ -2,7 +2,11 @@ import { FilterCheckbox } from "@/components/logs/checkbox/filter-checkbox";
 
-export const OutcomesFilter = () => {
+type OutcomesFilterProps = {
+  onOpenChange?: () => void;
```
it's not obvious what this does
does it close? does open?

chronark commented on apps/dashboard/app/(app)/apis/[apiId]/_overview/components/table/logs-table.tsx:
```diff
@@ -185,15 +185,15 @@ export const KeysOverviewLogsTable = ({ apiId, setSelectedLog, log: selectedLog
       emptyState={
-        <div className="w-full flex justify-center items-center h-full">
-          <Empty className="w-[400px] flex items-start">
+        <div className="w-full justify-center items-center h-[344px] md:h-full">
```
where's this magical 344px coming from?

chronark commented on apps/dashboard/components/logs/checkbox/filters-popover.tsx:
```diff
@@ -160,52 +199,96 @@ const FilterItem = ({
-          className={`flex w-full items-center px-2 py-1.5 justify-between rounded-lg group cursor-pointer
+    <>
+      {isMobile ? (
+        <Drawer.Nested>
```
wouldn't it be nice if we had a `Dialog` component that displays a popover or drawer automatically?
not saying we need to do this here, but that would make these things much cleaner

revogabe replied:
done

chronark commented on apps/dashboard/components/logs/datetime/suggestions.tsx:
```diff
@@ -72,10 +72,13 @@ export const DateTimeSuggestions = ({ className, options, onChange }: Suggestion
     >
-      <ScrollArea className="w-full h-full rounded-md max-h-[390px]" ref={scrollAreaRef}>
+      <ScrollArea
+        className="w-full h-full rounded-md max-h-[438px] md:max-h-[380px]"
```
why are we changing the max-h for md+ breakpoints?

chronark commented:
I think the nested drawers don't feel good when you're trying to exit them. I need to click away twice for the drawer to go away.
https://share.cleanshot.com/CRQmVk2b

chronark commented on apps/dashboard/components/ui/drawer.tsx:
```diff
@@ -0,0 +1,68 @@
+        {...contentProps}
+        ref={ref}
+        className={cn(
+          "bg-gray-1 border border-border flex flex-col fixed bottom-0 left-0 right-0 max-h-[82vh] rounded-t-xl drop-shadow-2xl",
```
please use `border-gray-6` according to https://engineering.unkey.com/design/colors

chronark commented on apps/dashboard/components/virtual-table/index.tsx:
```diff
@@ -79,10 +81,12 @@ export const VirtualTable = forwardRef<VirtualTableRef, VirtualTableProps<any>>(
 
     const containerClassName = cn(
-      "overflow-auto relative",
+      "overflow-auto relative pb-4",
```
why?

revogabe replied:
to create a slight space at the end of the scroll to give the impression that the results are over, without making the user think that it has crashed or that there is more content below

chronark reviewed (changes requested):
I love where this is going

let's fix a couple of small issues and then merge

revogabe commented:
Updates applied: implemented @chronark feedback, fixed bugs, added a responsive component that toggles between Drawer and Popover, and created a custom media‑query hook—no magic numbers, just mobile, tablet, or desktop.

chronark reviewed (approved):
great job
